TESTS/icb_arpack_cpp.cpp

```diff
@@ -94,106 +82,109 @@ void real_symmetric_runner(double const& tol_check) {
     throw std::domain_error("Error inside ARPACK routines");
   }
 
-  std::vector<a_int> select(ncv);
-  for (int i = 0; i < ncv; i++) select[i] = 1;
-
   arpack::neupd(rvec, arpack::howmny::ritz_vectors, select.data(), d.data(),
                 z.data(), ldz, sigma, workev.data(), arpack::bmat::identity, N,
-                arpack::which::largest_magnitude, nev, tol, resid.data(), ncv,
-                V.data(), ldv, iparam.data(), ipntr.data(), workd.data(),
+                ritz_option, nev, tol, resid.data(), ncv,
+                V.data(), ldv, iparam, ipntr, workd.data(),
                 workl.data(), lworkl, rwork.data(), info);
-
-  for (int i = 0; i < nev; ++i) {
-    Real rval = std::real(d[i]);
-    Real rref = static_cast<Real>(N - (nev - 1) + i);
-    Real reps = std::fabs(rval - rref);
-    Real ival = std::imag(d[i]);
-    Real iref = static_cast<Real>(N - (nev - 1) + i);
-    Real ieps = std::fabs(ival - iref);
-    std::cout << rval << " " << ival << " - " << rref << " " << iref << " - " << reps << " " << ieps << std::endl;
-
-    /*eigen value order: smallest -> biggest*/
-    if (reps > tol_check || ieps > tol_check) {
-      throw std::domain_error("Correct eigenvalues not computed");
+  if (info < 0) throw std::runtime_error("Error in neupd, info " + std::to_string(info));
+
+  if (ritz_option == arpack::which::largest_magnitude) {
+    for (int i = 0; i < nev; ++i) {
+      Real rval = std::real(d[i]);
+      Real rref = static_cast<Real>(N - (nev - 1) + i);
+      Real reps = std::fabs(rval - rref);
+      Real ival = std::imag(d[i]);
+      Real iref = -static_cast<Real>(N - (nev - 1) + i);
+      Real ieps = std::fabs(ival - iref);
+      std::cout << rval << " " << ival << " - " << rref << " " << iref << " - " << reps << " " << ieps << std::endl;
+
+      if (reps > tol_check || ieps > tol_check) throw std::domain_error("Correct eigenvalues not computed");
     }
+  } else if (ritz_option == arpack::which::largest_imaginary) {
+    for (int i = 0; i < nev; ++i) {
+      Real rval = std::real(d[i]);
+      Real rref = static_cast<Real>(nev - i);
+      Real reps = std::fabs(rval - rref);
+      Real ival = std::imag(d[i]);
+      Real iref = -static_cast<Real>(nev - i);
+      Real ieps = std::fabs(ival - iref);
+      std::cout << rval << " " << ival << " - " << rref << " " << iref << " - " << reps << " " << ieps << std::endl;
+
+      if (reps > tol_check || ieps > tol_check) throw std::domain_error("Correct eigenvalues not computed");
+    }
+  } else {
+    throw std::domain_error("The input Ritz option is not allowed in this test file.");
   }
+  std::cout << "------" << std::endl;
 }
 
 int main() {
   sstats_c();
 
   // arpack without debug
-  real_symmetric_runner<float>(1.);
-  real_symmetric_runner<double>(1.e-05);
+  real_symmetric_runner<float>(1., arpack::which::largest_magnitude);
+  real_symmetric_runner<double>(1.e-05, arpack::which::largest_magnitude);
 
```

Comment:
Possible to add 2 tests (float/double) with `arpack::which::largest_real` for instance? If not, no big deal.

Reply:
not a problem, will do